device_calendar/ios/Classes/SwiftDeviceCalendarPlugin.swift

```diff
@@ -211,8 +251,34 @@ public class SwiftDeviceCalendarPlugin: NSObject, FlutterPlugin {
             ekEvent!.startDate = startDate
             ekEvent!.endDate = endDate
             ekEvent!.calendar = ekCalendar!
+            
+            let recurrenceRuleArguments = arguments[recurrenceRuleArgument] as? Dictionary<String, AnyObject>
+            if (recurrenceRuleArguments != nil) {
+                let recurrenceFrequencyIndex = recurrenceRuleArguments![recurrenceFrequencyArgument] as? NSInteger
+                let totalOccurrences = recurrenceRuleArguments![totalOccurrencesArgument] as? NSInteger
+                let interval = recurrenceRuleArguments![intervalArgument] as? NSInteger
+                var recurrenceInterval = 1
+                let endDate = recurrenceRuleArguments![endDateArgument] as? NSNumber
+                let validFrequencyTypes = [EKRecurrenceFrequency.daily, EKRecurrenceFrequency.weekly, EKRecurrenceFrequency.monthly, EKRecurrenceFrequency.yearly]
+                let namedFrequency = validFrequencyTypes[recurrenceFrequencyIndex!]
+                
+                var recurrenceEnd:EKRecurrenceEnd?
+                if(endDate != nil) {
+                    recurrenceEnd = EKRecurrenceEnd(end: Date.init(timeIntervalSince1970: endDate!.doubleValue / 1000))
+                } else if(totalOccurrences != nil && totalOccurrences! > 0) {
+                    recurrenceEnd = EKRecurrenceEnd(occurrenceCount: totalOccurrences!)
+                }
+                
+                if(interval != nil && interval! > 1) {
+                    recurrenceInterval = interval!
+                }
+                ekEvent!.recurrenceRules = [EKRecurrenceRule(recurrenceWith: namedFrequency, interval: recurrenceInterval, end: recurrenceEnd)]
```

Comment:
Not sure if applicable but what would happen if all of the recurrence arguments are null (i.e. `endDate`, `totalOccurrences` and `interval`). Do we still want to create that rule?

Reply:
yes we would, only the frequency is mandatory. the rest are actually optional (FYI won't throw it they're nil)